addOns/authhelper/src/main/java/org/zaproxy/addon/authhelper/HistoryProvider.java

```diff
@@ -61,21 +90,67 @@ public HttpMessage getHttpMessage(int historyId)
         return null;
     }
 
+    /**
+     * The query is ordered DESCending so the List and subsequent processing should be newest
+     * message first.
+     */
+    List<Integer> getMessageIds(int first, int last, String value) {
+        if (!server) {
+            if (Model.getSingleton().getDb().getDatabaseServer()
+                    instanceof ParosDatabaseServer pdbs) {
+                server = true;
+                pds = pdbs;
+            } else {
+                LOGGER.warn("Unsupported Database Server.");
+            }
+        }
+        if (pds == null) {
+            return List.of();
+        }
+
+        try (Connection conn = pds.getSingletonConnection();
```

Comment:
Should just be called not managed, we don't want to close that connection.

Reply:
Yesterday you said instead of final block make it part of try with resource?

Reply:
Disregard, the connection is inlined.....now I get it.